### ISSUE TITLE
web-ext producing empty .zip files in v1.8.0

### ISSUE THREAD

**Issue by lmorchard** [mozilla/web-ext#789]
We're using the `web-ext` tool to build Snooze Tabs. [As of v1.8.0 we're getting an empty .zip file](https://github.com/bwinton/SnoozeTabs/issues/139).

With v1.7.0, things work as expected (487k zip file):
```
➜  tmp git clone https://github.com/bwinton/SnoozeTabs.git && cd SnoozeTabs
➜  SnoozeTabs git:(master) npm install && npm install web-ext@1.7.0 && npm run package && ls -al *.zip

# (...snip build process...)

Building web extension from /Users/lmorchard/devel/SnoozeTabs/dist
Your web extension is ready: snoozetabs-1.0.9.zip
-rw-r--r--  1 lmorchard  staff  487689 Feb  3 12:28 snoozetabs-1.0.9.zip
```

With v1.8.0, things do not work as expected (22 byte empty zip file):
```bash
➜  SnoozeTabs git:(master) rm *.zip && npm install web-ext@1.8.0 && npm run package && ls -al *.zip

# (...snip build process...)

Building web extension from /Users/lmorchard/devel/SnoozeTabs/dist
Your web extension is ready: snoozetabs-1.0.9.zip
-rw-r--r--  1 lmorchard  staff  22 Feb  3 12:25 snoozetabs-1.0.9.zip
```

Additional version info:
````
➜  SnoozeTabs git:(master) node --version && npm --version && web-ext --version
v6.2.0
3.8.9
1.8.0
````

**Comment by lmorchard** [comment]
Oh, and if it helps to reduce the test case, [this is how we call `web-ext` from an `npm` script](https://github.com/bwinton/SnoozeTabs/blob/master/package.json#L65):
```
npm run build && web-ext build -s dist -a .
```

**Comment by kumar303** [comment]
Oh crap, sorry about that. We started ignoring the artifacts directory by default in 1.8 but we didn't account for your `-a .` which is declaring the working directory as the artifacts directory.

@eight04 would you have some time to address this bug?

Here is a verbose log (using web-ext master) to illustrate the problem:
````
indigo:SnoozeTabs kumar$ web-ext --verbose build -s dist/ -a .
[program.js][debug] Getting version from the git revision
[program.js][info] Version: master-ec8d3ef7dcbf2c1d5459e5fc58a07f6cd1d33770
[util/file-filter.js][debug] Resolved path **/*.xpi with sourceDir /Users/kumar/tmp/SnoozeTabs/dist to /Users/kumar/tmp/SnoozeTabs/dist/**/*.xpi
[util/file-filter.js][debug] Resolved path **/*.zip with sourceDir /Users/kumar/tmp/SnoozeTabs/dist to /Users/kumar/tmp/SnoozeTabs/dist/**/*.zip
[util/file-filter.js][debug] Resolved path **/.* with sourceDir /Users/kumar/tmp/SnoozeTabs/dist to /Users/kumar/tmp/SnoozeTabs/dist/**/.*
[util/file-filter.js][debug] Resolved path **/.*/**/* with sourceDir /Users/kumar/tmp/SnoozeTabs/dist to /Users/kumar/tmp/SnoozeTabs/dist/**/.*/**/*
[util/file-filter.js][debug] Resolved path **/node_modules with sourceDir /Users/kumar/tmp/SnoozeTabs/dist to /Users/kumar/tmp/SnoozeTabs/dist/**/node_modules
[util/file-filter.js][debug] Resolved path **/node_modules/**/* with sourceDir /Users/kumar/tmp/SnoozeTabs/dist to /Users/kumar/tmp/SnoozeTabs/dist/**/node_modules/**/*
[util/file-filter.js][debug] Resolved path . with sourceDir /Users/kumar/tmp/SnoozeTabs/dist to /Users/kumar/tmp/SnoozeTabs/dist
[util/file-filter.js][debug] Resolved path **/* with sourceDir /Users/kumar/tmp/SnoozeTabs/dist to /Users/kumar/tmp/SnoozeTabs/dist/**/*
[cmd/build.js][info] Building web extension from /Users/kumar/tmp/SnoozeTabs/dist
[util/manifest.js][debug] Validating manifest at /Users/kumar/tmp/SnoozeTabs/dist/manifest.json
[util/file-filter.js][debug] Resolved path /Users/kumar/tmp/SnoozeTabs/dist/background.js with sourceDir /Users/kumar/tmp/SnoozeTabs/dist to /Users/kumar/tmp/SnoozeTabs/dist/background.js
[util/file-filter.js][debug] FileFilter: ignoring file /Users/kumar/tmp/SnoozeTabs/dist/background.js (it matched /Users/kumar/tmp/SnoozeTabs/dist/**/*)
[util/file-filter.js][debug] Resolved path /Users/kumar/tmp/SnoozeTabs/dist/lib with sourceDir /Users/kumar/tmp/SnoozeTabs/dist to /Users/kumar/tmp/SnoozeTabs/dist/lib
[util/file-filter.js][debug] FileFilter: ignoring file /Users/kumar/tmp/SnoozeTabs/dist/lib (it matched /Users/kumar/tmp/SnoozeTabs/dist/**/*)
[util/file-filter.js][debug] Resolved path /Users/kumar/tmp/SnoozeTabs/dist/icons with sourceDir /Users/kumar/tmp/SnoozeTabs/dist to /Users/kumar/tmp/SnoozeTabs/dist/icons
[util/file-filter.js][debug] FileFilter: ignoring file /Users/kumar/tmp/SnoozeTabs/dist/icons (it matched /Users/kumar/tmp/SnoozeTabs/dist/**/*)
[util/file-filter.js][debug] Resolved path /Users/kumar/tmp/SnoozeTabs/dist/manifest.json with sourceDir /Users/kumar/tmp/SnoozeTabs/dist to /Users/kumar/tmp/SnoozeTabs/dist/manifest.json
[util/file-filter.js][debug] FileFilter: ignoring file /Users/kumar/tmp/SnoozeTabs/dist/manifest.json (it matched /Users/kumar/tmp/SnoozeTabs/dist/**/*)
[util/file-filter.js][debug] Resolved path /Users/kumar/tmp/SnoozeTabs/dist/popup with sourceDir /Users/kumar/tmp/SnoozeTabs/dist to /Users/kumar/tmp/SnoozeTabs/dist/popup
[util/file-filter.js][debug] FileFilter: ignoring file /Users/kumar/tmp/SnoozeTabs/dist/popup (it matched /Users/kumar/tmp/SnoozeTabs/dist/**/*)
[cmd/build.js][info] Your web extension is ready: snoozetabs-1.0.9.zip
indigo:SnoozeTabs kumar$ unzip -l snoozetabs-1.0.9.zip
Archive:  snoozetabs-1.0.9.zip
warning [snoozetabs-1.0.9.zip]:  zipfile is empty
````

**Comment by kumar303** [comment]
@eight04 maybe the easiest fix is to *only* ignore the artifacts directory if it's *within* the source directory.

**Comment by kumar303** [comment]
@lmorchard thanks for your patience! The fix for this has been [released in 1.8.1](https://www.npmjs.com/package/web-ext) due to @eight04's speedy work \o/

**Comment by lmorchard** [comment]
Thanks for the quick fix!
